File: python/testing/test_scene_buffer.py
import tempfile
from pathlib import Path

import f3d


def test_scene_buffer():
    testing_dir = Path(__file__).parent.parent.parent / "testing"
    reference = Path(testing_dir) / "baselines/TestPythonSceneBuffer.png"
    output = Path(tempfile.gettempdir()) / "TestPythonSceneBuffer.png"
    points = testing_dir / "data/points.ply"

    engine = f3d.Engine.create(True)
    engine.window.size = 300, 300

    buf = Path(points).read_bytes()
    options = engine.options
    options["scene.force_reader"] = "PLYReader"
    engine.scene.add(buf, len(buf))

    img = engine.window.render_to_image()
    img.save(output)

    assert img.compare(f3d.Image(reference)) < 0.05

File: java/testing/TestSceneBuffer.java
import app.f3d.F3D.*;

import java.io.*;
import java.lang.String;

public class TestSceneBuffer {

  public static void main(String[] args) throws FileNotFoundException, IOException {
    Engine.autoloadPlugins();

    String testDataPath = args.length > 0 ? args[0] : ".";

    Engine engine = Engine.createNone();
    Scene scene = engine.getScene();

    char[] array = new char[256];
    FileReader input = new FileReader(testDataPath + "data/points.ply");
    int size = input.read(array);
    input.close();
    Options options = engine.getOptions();
    options.setAsString("scene.force_reader", "PLYReader");
    scene.addBuffer(new String(array).getBytes(), size);

    engine.close();
  }
}
